src/main/scala/workflow/GatherTransformer.scala

```diff
@@ -4,10 +4,10 @@ import org.apache.spark.rdd.RDD
 
 import scala.reflect.ClassTag
 
-private[workflow] class GatherTransformer[T] extends TransformerNode[Seq[T]] {
-  def transform(dataDependencies: Seq[_], fitDependencies: Seq[TransformerNode[_]]): Seq[T] = dataDependencies.map(_.asInstanceOf[T])
+private[workflow] class GatherTransformer[T] extends TransformerNode {
```

Comment:
I don't quite follow here what's going on with the fitDependencies - can you summarize?